embark.el

```diff
@@ -942,60 +961,80 @@ display actions and parameters are available."
+                                                bindings))))
+         (fmt (format "%%-%ds" (1+ max-width))))
+    (dolist (binding bindings)
+      (let ((cmd (caddr binding)))
+        (unless (embark--verbose-indicator-excluded-p cmd)
+          (let ((keys (format fmt (car binding)))
+                (doc (ignore-errors
+                       (propertize
+                        (car (split-string (documentation cmd) "\n"))
+                        'face 'embark-verbose-indicator-documentation))))
+            (insert keys (or doc "") "\n")))))))
+
+(defun embark--verbose-indicator-update (keymap target targets)
   "Update verbose indicator buffer.
-The arguments are the new KEYMAP, TARGET and OTHER-TARGETS."
+The arguments are the new KEYMAP, TARGET and other TARGETS."
   (with-current-buffer (get-buffer-create embark--verbose-indicator-buffer)
     (let* ((inhibit-read-only t)
            (bindings (car (embark--formatted-bindings keymap 'nested)))
-           (max-width (apply #'max (cons 0 (mapcar (lambda (x)
-                                                     (string-width (car x)))
-                                                   bindings))))
-           (fmt (format "%%-%ds" (1+ max-width))))
+           (ck (let ((ck (where-is-internal #'embark-cycle keymap)))
+                 (and ck (key-description (car ck))))))
       (setq-local cursor-type nil)
       (setq-local truncate-lines t)
       (setq-local buffer-read-only t)
       (erase-buffer)
-      (insert target)
-      (add-face-text-property (point-min) (point)
-                              'embark-verbose-indicator-title 'append)
-      (when (and other-targets (where-is-internal #'embark-cycle keymap))
-        (insert other-targets))
-      (insert "\n")
-      (dolist (binding bindings)
-        (let ((cmd (caddr binding)))
-          (unless (embark--verbose-indicator-excluded-p cmd)
-            (insert (format fmt (car binding))
-                    (or (ignore-errors
-                          (propertize
-                           (car (split-string (documentation cmd) "\n"))
-                           'face 'embark-verbose-indicator-documentation)) "")
-                    "\n"))))
+      (dolist (section embark-verbose-indicator-buffer-sections)
+        (cond ((stringp section) (insert section))
+              ((symbolp section)
+               (case section
+                 (newline (insert "\n"))
+                 (target
+                  (embark--verbose-indicator-insert-target (car target) (cdr target)))
+                 (other-targets
+                  (embark--verbose-indicator-insert-other-targets targets))
+                 (cycle (embark--verbose-indicator-insert-cycle-key ck))
+                 (bindings (embark--verbose-indicator-insert-bindings bindings))
+                 (t (when (fboundp section)
+                      (funcall section (cons target targets) bindings ck)))))))
```

Comment:
Good idea! I suggest to not use an explicit pcase here and dispatch only via funcall for conciseness:

`(funcall (intern (format "embark--verbose-indicator-insert-%s" section)) all the args...)`